src/main/scala/beam/agentsim/agents/BeamAgent.scala

```diff
@@ -83,23 +83,7 @@ trait BeamAgent[T] extends LoggingFSM[BeamAgentState, T] with Stash {
       case None =>
         ""
     }
-    s"$tickStr${triggerStr}State:$stateName ${logPrefix()}$msg"
-  }
-
-  def logInfo(msg: String): Unit = {
-    log.info(s"${logWithFullPrefix(msg)}")
-  }
-
-  def logWarn(msg: String): Unit = {
-    log.warning(s"${logWithFullPrefix(msg)}")
-  }
-
-  def logError(msg: String): Unit = {
-    log.error(s"${logWithFullPrefix(msg)}")
-  }
-
-  def logDebug(msg: String): Unit = {
-    log.debug(s"${logWithFullPrefix(msg)}")
+    s"$tickStr${triggerStr}State:$stateName ${logPrefix()}"
```

Comment:
Previously was logging something. Now creates a string and nothing happens. Looks like a dangling change here.

Reply:
No, he ripped it out in favor of explicit `log.debug`. I'm not sure that should be the path here, though. If it's the creation of `msg: String` that is the issue then how about making it a byval parameter:

`def logDebug(msg: => String): Unit = { ....`

Then you can keep the prefix portion encapsulated, and not something users have to remember to use when calling log. You could even make it a variable length argument if you want to avoid string building by the callers. The primary point here is that this change now forces users to have to know/remember to add the prefix to their logs.